### ISSUE TITLE
Mark message as unread not working

### ISSUE THREAD

**Issue by olodar** [RocketChat/Rocket.Chat#29741]
### Description:

When I'm trying to mark message as unread, RC opens home page instead.

![Screenshot from 2023-07-06 10-20-09](https://github.com/RocketChat/Rocket.Chat/assets/32993068/d84f77ce-6438-4af9-929b-ed44bb4698e7)
![Screenshot from 2023-07-06 10-20-20](https://github.com/RocketChat/Rocket.Chat/assets/32993068/64376679-1147-4765-9356-c79e4152ee77)

### Steps to reproduce:

1. Try to mark any message as unread

### Expected behavior:

Message should mark as unread

### Actual behavior:

It forwards me to home screen instead

### Server Setup Information:

- Version of Rocket.Chat Server: 6.2.9
- Operating System: Ubuntu 22.04
- Deployment Method: tar
- Number of Running Instances: 1
- DB Replicaset Oplog: -
- NodeJS Version: v14.21.3
- MongoDB Version: 6.0.6

### Client Setup Information

- Desktop App or Browser Version: Chromium 111.0.5563.65
- Operating System: Ubuntu 22.04

**Comment by dhruv33333** [comment]
Hey, I have looked into this, this is fixable but there is one issue that I thought of. 
What should happen when the user "marks as unread" the channel which is currently opened?
After removing the redirection, currently it is just showing the loading state continuously.

**Comment by olodar** [comment]
> What should happen when the user "marks as unread" the channel which is currently opened?

I think message should stay unread until you reopen channel/thread.

**Comment by Ninad3721** [comment]
The feature is working as expected it should redirect you to the Home page and the channel whose message is unread , its name gets bold which means it contains unread message.

**Comment by hugocostadev** [comment]
@Ninad3721 is correct... @olodar the channel/DM name it's bold there? 

Can you test again with different channels? Thanks

---
**Questions? Help needed? Feature Requests?** 

- Join our [Open Server](https://open.rocket.chat/) in the #support channel and feel free to raise a question
- Join our [Community Forum](https://forums.rocket.chat/) and search/create a post there
- Feature Request: Open an issue in [Feature Request](https://github.com/RocketChat/feature-requests) repository

**Comment by olodar** [comment]
> the channel/DM name it's bold there?

Yes, it is works how [Ninad3721](https://github.com/Ninad3721) described.
